### PR TITLE
documentation: new structure and added backend architecture docs

### PR DESCRIPTION
**Pull request by joka**
Motivation:

Working on the architectue documentation i noticed the docs are quite confusing, many parts missing and some information duplicated. So I also started some restructering without changing to much.

Future tasks:

We should make a  clear structure targeting the different documentation  users (Core/Extensions developers, Designers, Integrators/Process designers, sysadmins). To improve the documentation in the long run we should add documentation to thedefinition of done. Whats is missing at all is the description of domain specific language, the basic Modelling for particaption processes and a narrative documentation how to work with the backend\frontend framework.

Content Changes:

- common introduction and backend introduction (needs refactoring)
- add contributed code with git (formerly git guideline)
- add backend architecture diagramms and updated module overview
- add introduction to code refactoring and testing

Structure:

- development  (developer core)

  - development installation
  - development tasks
  - code styles guides
  - refactore guide
  - contribute code with git guide

- Administration tasks (sysadmin)    
- backend (developer core/extensions)
- frontend Javascript (developer core/extensions)
- frontend REST-API  (external developer)
- Project specific ( external developer)

### CODE REVIEW

**Comment by xi** [comment]
I have a build error:

```
$ bin/ad_build_doc 
../bin/sphinx-build -b html -d build/doctrees  -an . build/html
Running Sphinx v1.3.5

Extension error:
Could not import extension sphinxcontrib.blockdiag (exception: No module named 'sphinxcontrib')
Makefile:54: recipe for target 'html' failed
make: *** [html] Error 1
```

**Comment by joka** [comment]
sorry,  i was using a GUI for interactive rebase that ended in quite a mess

**Comment by joka** [comment]
Build error fixed

File: docs/backend/architecture.rst [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+adhocracy_sample
+   examples how to customize resource/sheet types
+
+adhocracy_frontend
+   framework for the javascript frontend
+
+adhocracy_xyz
+   Backend extensions for project spezific application
+
+xzy
```
Comment (2e2a) [inline docs/backend/architecture.rst]: typo: "xyz"

File: docs/backend/modules.rst [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    should lay at one single point of code (packages/modules in this case),
+    see also :doc:`../coding_guides/refactore_guidelines`.
+
+*layer*
+    loosly group of modules that follow these rules:
+    adhocracy_core.evolution
+    * must not import from upper layer
+    * should not import from same layer
+    * may import interfaces from all layers
+    * may import from lower layer
```
Comment (2e2a) [inline docs/backend/modules.rst]: Remove adhocracy_core.evolution

File: docs/backend/overview.rst [hunk 3]
```diff
@@ -0,0 +1,121 @@
+Overview
+========
+
+The adhocracy backend is python framework to build a REST-API *backend* for
+cms-like applications.
```
Comment (2e2a) [inline docs/backend/overview.rst]: typo: "a python framework"

File: docs/backend/overview.rst [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+`Resources`:
+
+    - composed by a set of sheets (see :doc:`../attic/instance_modelling_example`)
+    - runtime adding/removing of sheets possible
+    - `open/close principal <https://en.wikipedia.org/wiki/Open/closed_principle>`_ for resource modelling
+
+`Sheets`:
+
+    - interface to for a specific resource behaviour:
+      - api methods
```
Comment (2e2a) [inline docs/backend/overview.rst]: typo: Remove "to"

Syntax: Not shown as list

File: docs/development/coding_guides/python_guidelines.rst [hunk 5]
```diff
@@ -1,14 +1,20 @@
...
-  * use `pytest <http://pytest.org/>`_ fixtures, functional tests have the `functional` marker, integration are using a fixture called `integration`.
+* use `pytest <http://pytest.org/>`_ fixtures to mock/create dependencies, functional tests have the `functional` marker, integration are using a fixture called `integration`.
+* Test driven development with functional/integration and unit test (should)
+
+    concept: http://en.wikipedia.org/wiki/Test-driven_development
+
+    1. write function/integration test
+    2. write unit test (simplest statement first)
+    3. switch between writing code and change/extend tests until all test pass
+    4. refactor (:doc:`refactore_guidelines`)
```
Comment (2e2a) [inline docs/development/coding_guides/python_guidelines.rst]: typo in link: refactor_guidelienes

**Comment by pallix** [comment]
Nice!

**Pull request merged**